snuba/datasets/sessions.py

```diff
@@ -0,0 +1,263 @@
+            [
+                ("session_id", UUID()),
+                ("distinct_id", UUID()),
+                ("seq", UInt(64)),
+                ("org_id", UInt(64)),
+                ("project_id", UInt(64)),
+                ("retention_days", UInt(16)),
+                ("duration", UInt(32)),
+                ("status", UInt(8)),
+                ("errors", UInt(16)),
+                # note here that deleted is really only an indicator so we can
+                # figure out which values can be discarded.  Deleting of values
+                # does not update the materializations.
+                ("deleted", UInt(8)),
+                ("received", DateTime()),
+                ("started", DateTime()),
+                ("release", LowCardinality(String())),
+                ("environment", LowCardinality(String())),
+            ]
+        )
+
+        write_schema = MergeTreeSchema(
+            columns=all_columns,
+            local_table_name=WRITE_LOCAL_TABLE_NAME,
+            dist_table_name=WRITE_DIST_TABLE_NAME,
+            order_by="(org_id, project_id, toDate(started), session_id, cityHash64(toString(session_id)))",
+            partition_by="(toMonday(received))",
+            sample_expr="cityHash64(toString(session_id))",
+            settings={"index_granularity": 16384},
+        )
+
+        read_columns = ColumnSet(
+            [
+                ("org_id", UInt(64)),
+                ("project_id", UInt(64)),
+                ("started", DateTime()),
+                ("release", LowCardinality(String())),
+                ("environment", LowCardinality(String())),
+                (
+                    "duration_quantiles",
+                    AggregateFunction("quantilesIf(0.5, 0.9)", UInt(32), UInt(8)),
+                ),
+                ("uniq_sessions", AggregateFunction("countIf", UUID(), UInt(8))),
+                ("uniq_users", AggregateFunction("uniqIf", UUID(), UInt(8))),
+                ("uniq_sessions_crashed", AggregateFunction("countIf", UUID(), UInt(8))),
+                (
+                    "uniq_sessions_abnormal",
+                    AggregateFunction("countIf", UUID(), UInt(8)),
+                ),
+                ("uniq_sessions_errored", AggregateFunction("uniqIf", UUID(), UInt(8))),
+                ("uniq_users_crashed", AggregateFunction("uniqIf", UUID(), UInt(8))),
+                ("uniq_users_abnormal", AggregateFunction("uniqIf", UUID(), UInt(8))),
+                ("uniq_users_errored", AggregateFunction("uniqIf", UUID(), UInt(8))),
+            ]
+        )
+        read_schema = AggregatingMergeTreeSchema(
+            columns=read_columns,
+            local_table_name=READ_LOCAL_TABLE_NAME,
+            dist_table_name=READ_DIST_TABLE_NAME,
+            prewhere_candidates=["project_id", "org_id"],
+            order_by="(org_id, project_id, release, environment, started)",
+            partition_by="(toMonday(started))",
+            settings={"index_granularity": 16384},
```

Comment:
We should be able to drop this granularity down drastically. Like, 256 drastically.